pkgs/development/libraries/glib/gdb_script.patch

```diff
@@ -0,0 +1,46 @@
+diff --git a/glib/meson.build b/glib/meson.build
+index b2dd569e1..b013991a1 100644
+--- a/glib/meson.build
++++ b/glib/meson.build
+@@ -513,8 +513,7 @@ configure_file(
+ )
+ 
+ install_data('glib_gdb.py',
+-  install_dir : glib_pkgdatadir / 'gdb',
+-  install_tag : 'devel',
++  install_dir : get_option('datadir') / 'glib-2.0' / 'gdb',
+ )
+ 
+ # This is needed to make gdb find glib_gdb.py
+@@ -541,7 +540,7 @@ configure_file(
+   input: 'libglib-gdb.py.in',
+   output: 'libglib-2.0.so.@0@-gdb.py'.format(library_version),
+   configuration: gdb_conf,
+-  install_dir: gdb_install_dir,
++  install_dir: glib_libdir,
+   install_tag: 'devel',
+   install: gdb_install,
+ )
+diff --git a/gobject/meson.build b/gobject/meson.build
+index 78b732bf3..cf1e6204f 100644
+--- a/gobject/meson.build
++++ b/gobject/meson.build
+@@ -176,8 +176,7 @@ gobject_query = executable('gobject-query', 'gobject-query.c',
+   dependencies : [libglib_dep, libgobject_dep])
+ 
+ install_data('gobject_gdb.py',
+-  install_dir : glib_pkgdatadir / 'gdb',
+-  install_tag : 'devel',
++  install_dir : get_option('datadir') / 'glib-2.0' / 'gdb',
+ )
+ gdb_conf = configuration_data()
+ gdb_conf.set('datadir', glib_datadir)
+@@ -185,7 +184,7 @@ configure_file(
+   input: 'libgobject-gdb.py.in',
+   output: 'libgobject-2.0.so.@0@-gdb.py'.format(library_version),
+   configuration: gdb_conf,
+-  install_dir: gdb_install_dir,
++  install_dir: glib_libdir,
```

Comment:
Is not the current one (`/nix/store/3xsbahrqqc4fc3gknmjj9j9687n4hiz0-glib-2.80.0/share/gdb/auto-load/nix/store/3xsbahrqqc4fc3gknmjj9j9687n4hiz0-glib-2.80.0/lib/libglib-2.0.so.0.8000.0-gdb.py`) the correct location? For example, Arch also uses [`/usr/share/gdb/auto-load/usr/lib/libglib-2.0.so.0.8000.2-gdb.py`](https://archlinux.org/packages/core/x86_64/glib2/files/).

It sounds more like the issue is that the file refers to `$out` output, while the scripts are in `$dev`.

Reply:
If I understand [the documentation](https://sourceware.org/gdb/current/onlinedocs/gdb.html/objfile_002dgdbdotext-file.html#objfile_002dgdbdotext-file)  correctly, scripts are looked in:
- the folder of the .so file. this is how the script of libstdc++ works.
- the global folder printed by `show data-directory`. It seems to be ${pkgs.gdb}/share/gdb. Meson installs to /nix/store/3xsbahrqqc4fc3gknmjj9j9687n4hiz0-glib-2.80.0/share/gdb/auto-load/ in the hope of writing to this directory, which works on arch but not for us.